ncm-grub/src/main/perl/grub.pm

```diff
@@ -74,534 +77,543 @@ sub grubbyArgsOptions {
+                              qr/^password $val$/,
+                              "password $val\n",
+                              $self->main_section_offset($grub_fh),
+                              );
+}
+
+# get the position for the main section
+# ie. after the header comments
+sub main_section_offset {
+    my ($self, $fh) = @_;
+    my ($start, $end) = $fh->get_header_positions();
+    return $start == -1 ? BEGINNING_OF_FILE : (SEEK_SET, $end);
+}
+
 ##########################################################################
 sub Configure {
 ##########################################################################
-  my ($self,$config)=@_;
-
-  my $grubby='/sbin/grubby';
-  my $prefix='/boot';
-  if ($config->elementExists("/software/components/grub/prefix")) {
-    $prefix = $config->getValue("/software/components/grub/prefix");
-  }
-  my $kernelname='vmlinuz';
-  my $kernelversion = '';
-  if ( $config->elementExists(KERNELPATH_CONFIG) ) {
-      $kernelversion = $config->getValue(KERNELPATH_CONFIG);
-  };
-  my $fulldefaultkernelpath;
-  # An undefined kernel version or an empty string are treated equally
-  if ( !$kernelversion ) {
-      $self->debug(1,"No kernel version defined: default kernel will not be set");
-  } else {
-      $fulldefaultkernelpath=$prefix.'/'.$kernelname.'-'.$kernelversion;
-  }
-
-  my $cons = undef;
-  my $consolepath = "/hardware/console/serial";
-  my $serialcons  = undef;
-  my $newserial = "";
-  my $newterminal = "";
-  if ($config->elementExists($consolepath)) {
-    my %consnode = $config->getElement($consolepath)->getHash();
-
-    my $unit = "0";
-    if (exists $consnode{"unit"}) {
-      $unit = $consnode{"unit"}->getValue();
-    }
+    my ( $self, $config ) = @_;
 
-    my $speed = "9600";
-    if (exists $consnode{"speed"}) {
-      $speed = $consnode{"speed"}->getValue();
+    my $grubby = '/sbin/grubby';
+    my $prefix = '/boot';
+    if ( $config->elementExists("/software/components/grub/prefix") ) {
+        $prefix = $config->getValue("/software/components/grub/prefix");
+    }
+    my $kernelname    = 'vmlinuz';
+    my $kernelversion = '';
+    if ( $config->elementExists($PATH_KERNEL_VERSION) ) {
+        $kernelversion = $config->getValue($PATH_KERNEL_VERSION);
     }
+    my $fulldefaultkernelpath;
 
-    my $word = "8";
-    if (exists $consnode{"word"}) {
-      $word = $consnode{"word"}->getValue();
+    # An undefined kernel version or an empty string are treated equally
+    if ( !$kernelversion ) {
+        $self->debug( 1, "No kernel version defined: default kernel will not be set" );
+    } else {
+        $fulldefaultkernelpath = $prefix . '/' . $kernelname . '-' . $kernelversion;
     }
 
-    my $parity = "n";
-    if (exists $consnode{"parity"}) {
-      $parity = $consnode{"parity"}->getValue();
+    my $grub_fh = CAF::FileEditor->new( $GRUB_CONF,
+        owner => "root",
+        group => "root",
+        mode  => 0400,
+        log   => $self );
```

Comment:
please be fully aware that you will log any changes to the grub config file (when level is high enough), incl the password (hash).
this "feature" should be at least mentioned in the pod